pkg/catalog/routes.go

```diff
@@ -435,3 +435,28 @@ func (mc *MeshCatalog) routesFromRules(rules []target.TrafficTargetRule, traffic
 
 	return routes, nil
 }
+
+// GetServicesForServiceAccounts returns a list of services corresponding to a list service accounts
+//	TODO: Consider merging this function and mc.GetServicesForServiceAccount in future (#2038)
+func (mc *MeshCatalog) GetServicesForServiceAccounts(saList []service.K8sServiceAccount) []service.MeshService {
+	serviceMap := map[service.MeshService]int{}
```

Comment:
Should we use a set instead of a map ?

Reply:
An alternative could be 👍 
```suggestion
	serviceMap := map[service.MeshService]bool{}
```

This will need lesser memory as compared to int. My preference would still be a set